cpp/src_prims/sparse/selection/knn.cuh

```diff
@@ -434,8 +378,12 @@ class sparse_knn_t {
     dist_config.allocator = allocator;
     dist_config.stream = stream;
 
-    raft::sparse::distance::pairwiseDistance(batch_dists, dist_config,
-                                             get_pw_metric(), metricArg);
+    if (raft::sparse::distance::supportedDistance.find(metric) ==
```

Comment:
I like the introduction of the explicit set for supported distances. For a follow-on / future PR, we might want to consider just using a hash map to map the distance enum type to its “distances_t” implementation, which will allow us to get rid of the switch statement altogether.